Add promu support to build binary

CharlesJUDITH opened the pull request:
- Modify Makefile for promu
- Add VERSION file
- Add promu.yml

Commit:
Add promu support to build binary

- Modify Makefile for promu
- Add VERSION file
- Add promu.yml

CharlesJUDITH commented:
I can add the .travis.yml in the same commit.

mattbostock commented on .promu.yml:
```diff
@@ -0,0 +1,44 @@
+tarball:
+    files:
+        - LICENSE
+        - NOTICE
```
`NOTICE` does not exist.

mattbostock commented on .promu.yml:
```diff
@@ -0,0 +1,44 @@
+          path: .
+    flags: -a -tags netgo
+    ldflags: |
+        -X {{repoPath}}/vendor/github.com/prometheus/common/version.Version={{.Version}}
```
None of these would have any effect currently, though I'd welcome changes to make use of the `version` package.

Currently, only `main.revision` is set during compile time.

mattbostock commented on Makefile:
```diff
@@ -1,7 +1,57 @@
+
+assets:
+	@echo ">> writing assets"
+	-@$(GO) get -u github.com/jteeuwen/go-bindata/...
```
There are no assets to build.

mattbostock commented on Makefile:
```diff
@@ -1,7 +1,57 @@
+	GOARCH=$(subst x86_64,amd64,$(patsubst i%86,386,$(shell uname -m))) \
+	$(GO) get -u github.com/prometheus/promu
+
+proto:
```
This target should be removed.

mattbostock commented on Makefile:
```diff
@@ -1,7 +1,57 @@
-.PHONY: build test
+GO    := GO15VENDOREXPERIMENT=1 go
+PROMU := $(GOPATH)/bin/promu
+pkgs   = $(shell $(GO) list ./... | grep -v -E '/vendor/|/ui')
```
There is no `ui` directory (nor any plan to add one).

mattbostock reviewed (changes requested):
Thanks for the PR! I agree with using promu as it'll make this project more consistent with other Prometheus projects.

I've commented on some minor nits. Please push the Travis changes for review (same commit or separate is fine).

Commit:
Refactoring for promu settings

- Remove useless targets in Makefile
- Fix grep in .promu.yml
- Add .travis.yml

CharlesJUDITH commented:
Regarding the travis build you'll need something like:
```yml
sudo: required

services:
- docker

language: go

go:
- 1.7.x
- 1.8.x
- master

matrix:
  allow_failures:
  - go: master

script:
- make style test

before_deploy:
- make promu
- promu crossbuild
- promu crossbuild tarballs
- promu checksum .tarballs

deploy:
  provider: releases
  api_key:
    secure: $GITHUB_OAUTH_TOKEN
  file_glob: true
  file: .tarballs/*
  skip_cleanup: true
  on:
    tags: true
```
There is a useful documentation here: https://docs.travis-ci.com/user/build-stages/deploy-github-releases/

CharlesJUDITH commented:
You can see the releases from my fork https://github.com/CharlesJUDITH/alertmanager2es/releases

CharlesJUDITH commented:
This issue can be closed
